### PR TITLE
Bullets

### LINKED ISSUE #8: Add bullet mechanic
Add bullets. Bullets move in a linear direction with constant velocity, and collide with the environment and characters. There should be different types of bullets: bullets that will bounce off environment colliders, bullets that will explode on collision, maybe bullets with some sort of tracking that will deviate towards its target... For now, bullets that collide and bounce is enough.

### PR DESCRIPTION
**Pull request by dmonzonis**
Add bouncing and non-bouncing bullets. Velocity should be given to them on instantiation, as they will maintain constant speed until destroyed.
Resolves #8

### CODE REVIEW

File: scripts/bullet.gd [hunk 1]
```diff
@@ -0,0 +1,33 @@
+extends KinematicBody2D
+
+var bounces # If true, will bounce on collision with environment instead of exploding
+var velocity
+
```
Comment (Elkasitu) [inline scripts/bullet.gd:5]: export both bounces and velocity?

File: scripts/bullet.gd [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+			velocity -= 2 * velocity.dot(normal) * normal
+		else:
+			queue_free()
+
+# Signal received when the timer reaches lifeTime. Destroys the bullet.
+func _on_timer_timeout():
+	queue_free()
+
+# Utility function to initialize the projectile, setting its velocity and starting the timer
+func initialize(position, direction, speed, lifeTime = 2, _bounces = false):
```
Comment (Elkasitu) [inline scripts/bullet.gd:25]: maybe rename to something like spawn instead of initialize so as to not confuse it with _init

**Review by Elkasitu — APPROVED** [review]
all good, all comments are optional and not mandatory

File: scripts/bullet.gd (hunk 1 above)
Comment (dmonzonis) [reply scripts/bullet.gd:5]: bullets will be always created from code so there's no need

**Pull request merged**